File: frontend/lib/models/order.dart
class Order {
  final int orderId;
  final DateTime orderDate;
  final String customerName;
  final String supplierName;
  final double totalAmount;
  final String status;
  final List<OrderDetail> details;

  Order({
    required this.orderId,
    required this.orderDate,
    required this.customerName,
    required this.supplierName,
    required this.totalAmount,
    required this.status,
    required this.details,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      orderId: json['order_id'],
      orderDate: DateTime.parse(json['order_date']),
      customerName: json['customer_name'],
      supplierName: json['supplier_name'],
      totalAmount: json['total_amount'].toDouble(),
      status: json['status'],
      details: (json['details'] as List)
          .map((detail) => OrderDetail.fromJson(detail))
          .toList(),
    );
  }
}

class OrderDetail {
  final int productId;
  final String productName;
  final int quantity;
  final double price;

  OrderDetail({
    required this.productId,
    required this.productName,
    required this.quantity,
    required this.price,
  });

  factory OrderDetail.fromJson(Map<String, dynamic> json) {
    return OrderDetail(
      productId: json['product_id'],
      productName: json['product_name'],
      quantity: json['quantity'],
      price: json['price'].toDouble(),
    );
  }
}

File: backend/app/schemas/order.py
from pydantic import BaseModel, Field
from typing import List, Optional
from datetime import date
from decimal import Decimal

class OrderDetailBase(BaseModel):
    product_id: int
    quantity: int = Field(..., gt=0)

class OrderDetail(OrderDetailBase):
    order_detail_id: int
    product_name: str
    unit_price: Decimal
    total_price: Decimal

    class Config:
        from_attributes = True

class OrderBase(BaseModel):
    order_date: date
    supplier_id: int
    customer_id: int
    details: List[OrderDetailBase]

class OrderCreate(OrderBase):
    pass

class OrderUpdate(BaseModel):
    order_date: Optional[date] = None
    supplier_id: Optional[int] = None
    details: Optional[List[OrderDetailBase]] = None

class OrderResponse(BaseModel):
    order_id: int
    order_date: date
    supplier_id: int
    supplier_name: str
    customer_id: int
    customer_name: str
    total_items: int
    total_quantity: int
    total_amount: Decimal
    amount_paid: Decimal
    status: str

    class Config:
        from_attributes = True

class OrderSummary(BaseModel):
    order_id: int
    order_date: date
    supplier_name: str
    customer_name: str
    total_amount: Decimal

    class Config:
        from_attributes = True

class OrderStatusSummary(BaseModel):
    order_id: int
    total_items: int
    total_quantity: int
    total_amount: Decimal
    amount_paid: Decimal
    status: str

    class Config:
        from_attributes = True